loopy/check.py

```diff
@@ -421,8 +426,10 @@ def map_subscript(self, expr):
                             len(subscript), len(shape)))
 
             try:
+                context, assumptions = isl.align_two(context,
+                                                     self.kernel.assumptions)
```

Comment:
We could probably save some time by doing this once, upon entry into the mapper.

Reply:
Done. (Now it's done once for every instruction, still not ideal but OK)